processing/src/main/java/io/druid/query/topn/TopNMapFn.java

```diff
@@ -60,7 +60,9 @@ public TopNMapFn(
       return resultBuilder.build();
     }
     finally {
-      topNAlgorithm.cleanup(params);
+      if (params != null) {
```

Comment:
Usually null pointers are handled near the dereferencing point to help in case future developers call the method in another way and run into similar issues. Can the null checked be moved to the cleanup() method?